README.md

```diff
@@ -145,7 +183,12 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+To run main.cpp, type these commands:
```

Comment:
First, thank you for providing both positive and negative tests. The thing that stops them getting full marks is that they don't actually do any testing: they just report facts to the console, and a poor human has to match them against some strings in the code. A test would compare the result of each parse against the expected value, and report when they didn't match.

# -2

You `determineValidity` function is a nice piece of C++ (and I don't say that lightly: I've read enough C++ in the last two days to last a lifetime, so it has to be really good for me to appreciate it.)  However, you aren't actually implementing a state machine. In fact, deleting the variable `state` makes no difference. I was hoping to see a pure table-based approach:

~~~ c++
typedef int state;
typedef std::map<std::string, state> transitions;
typedef std::map<state, transitions> state_table;

const int END = 99;

state_table states = {
  { 0, {{ "the", 1 }}},
	{ 1, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 2, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 3, {{ "ate", 4},  {"ran",    4}}},
	{ 4, {{ "slowly", 5}, { "noisily", 5}, {"EOI", END}}},
	{ 5, {{ "EOI", END }}}
};


int getNextState(int currentState, string nextWord) {
	transitions t = states[currentState];
	if (t.count(nextWord))
		return t[nextWord];
	else
		return -1;
}
~~~

I normally deduct 6 for not using a table, but given how nice your code was to read:

# -3